pkg/jsonCrawler/json-crawler.go

```diff
@@ -0,0 +1,189 @@
+	anonymizeItem     anonymizeFunc
+	ctx               context.Context
+	analyzeTemplate   *types.AnalyzeTemplate
+	anonymizeTemplate *types.AnonymizeTemplate
+}
+
+//New JSONCrawler
+func New(ctx context.Context, analyze analyzeFunc, anonymize anonymizeFunc, analyzeTemplate *types.AnalyzeTemplate, anonymizeTemplate *types.AnonymizeTemplate) *JSONCrawler {
+	return &JSONCrawler{
+		analyzeItem:       analyze,
+		anonymizeItem:     anonymize,
+		ctx:               ctx,
+		analyzeTemplate:   analyzeTemplate,
+		anonymizeTemplate: anonymizeTemplate,
+	}
+}
+
+// ScanJSON scan the json in DFS to get to all the nodes
+func (jsonCrawler *JSONCrawler) ScanJSON(schemaMap map[string]interface{}, valuesMap map[string]interface{}) error {
+	err := checkIfEmptyMap(schemaMap, valuesMap)
+	if err != nil {
+		return err
+	}
+
+	for key, val := range schemaMap {
+		switch concreteVal := val.(type) {
+		case map[string]interface{}:
+			err := jsonCrawler.scanIfNotEmpty(valuesMap, key, val, "map")
+			if err != nil {
+				return err
+			}
+
+		case []interface{}:
+			err := jsonCrawler.scanIfNotEmpty(valuesMap, key, val, "array")
+			if err != nil {
+				return err
+			}
+		default:
+			newVal, err := checkIfKeyExistInMap(valuesMap, key)
+			if err != nil {
+				return err
+			}
+			anonymizedVal, err := jsonCrawler.analyzeAndAnonymizeJSON(fmt.Sprint(newVal), fmt.Sprint(concreteVal))
+			if err != nil {
+				return err
+			}
+			valuesMap[key] = anonymizedVal
+		}
+	}
+	return nil
+}
+
+func (jsonCrawler *JSONCrawler) scanArray(schemaArray []interface{}, valuesArray []interface{}) error {
+	err := checkIfEmptyArray(schemaArray, valuesArray)
+	if err != nil {
+		return err
+	}
+
+	i := 0
+	for j := range valuesArray {
+		if len(schemaArray) > 1 {
+			i = j
+		}
+
+		val := schemaArray[i]
+		switch concreteVal := val.(type) {
+		case map[string]interface{}:
+			err := jsonCrawler.ScanJSON(val.(map[string]interface{}), valuesArray[j].(map[string]interface{}))
+			if err != nil {
+				return err
+			}
+		case []interface{}:
+			err := jsonCrawler.scanArray(val.([]interface{}), valuesArray[j].([]interface{}))
+			if err != nil {
+				return err
+			}
+		default:
+			newVal, err := jsonCrawler.analyzeAndAnonymizeJSON(fmt.Sprint(valuesArray[j]), fmt.Sprint(concreteVal))
+			if err != nil {
+				return err
+			}
+			valuesArray[j] = newVal
+		}
+	}
+
+	return nil
+}
+
+func (jsonCrawler *JSONCrawler) scanIfNotEmpty(valuesMap map[string]interface{}, key string, val interface{}, valType string) error {
+	newVal, err := checkIfKeyExistInMap(valuesMap, key)
+	if err != nil {
+		return err
+	}
+	if valType == "map" {
+		return jsonCrawler.ScanJSON(val.(map[string]interface{}), newVal.(map[string]interface{}))
+	}
+
+	return jsonCrawler.scanArray(val.([]interface{}), newVal.([]interface{}))
+}
+
+func (jsonCrawler *JSONCrawler) analyzeAndAnonymizeJSON(val string, field string) (string, error) {
+	match, err := regexp.MatchString("<[A-Z]+(_*[A-Z]*)*>", field)
```

Comment:
move the hard coded regex to the top of the file and give it a meaningful name/comment regarding what it finds.
e.g.
const namesRegex="...."